docs/rootfs-and-kernel-setup.md

```diff
@@ -0,0 +1,137 @@
+# Creating Custom rootfs and kernel Images
+
+## Creating a kernel Image
+
+Currently, Firecracker supports only uncompressed, ELF kernel images. You can
+build an uncompressed Linux kernel image with:
+
+```bash
+make vmlinux
+```
+
+Here's a quick step-by-step guide to building your own kernel that Firecracker
+can boot:
+1. Get the Linux source code:
+
+   ```bash
+   git clone https://github.com/torvalds/linux.git linux.git
+   cd linux.git
+   ```
+   
+2. Check out the Linux version you want to build (e.g. we'll be using v4.20
+   here):
+   
+   ```bash
+   git checkout v4.20
+   ```
+   
+3. You will need to configure your Linux build. You can start from
+   [our recommended config](../resources/microvm-kernel-config) - just copy
+   it to `.config` (under the Linux sources dir). You can make interactive
+   config adjustments using:
+   
+   ```bash
+   make menuconfig
+   ```
+   
+   *Note*: there are many ways of building a kernel config file, other than
+   `menuconfig`. You are free to use whichever one you choose.
+   
+4. Build the uncompressed kernel image:
+
+   ```bash
+   make vmlinux
+   ```
+
+5. Upon a successful build, you can find the uncompressed kernel image under
+   `./vmlinux`.
+
+
+## Creating a rootfs Image
+
+A rootfs image is just a file system image, that hosts at least an init
+system. For instance, our getting started guide uses an EXT4 FS image with
+OpenRC as an init system. Note that, whichever file system you choose to use,
```

Comment:
```suggestion
`OpenRC` as an init system. Note that, whichever file system you choose to use,
```

(you use `OpenRC` between backtics further in the file, so it's also a matter of consistency)

Reply:
Removed backticks.